src/main/resources/org/jenkinsci/plugins/form-element-path.js

```diff
@@ -165,7 +165,9 @@ Behaviour.addLoadEvent(function(){
                 function child(v,i) {
                     var suffix=null;
                     var newKey=key;
-                    if (v.getAttribute("type")=="radio"){
+                    if (v.getAttribute("name")=="publisher") {
+                        suffix = v.getAttribute("descriptorId").split(".").pop()
```

Comment:
This presume [hetero-list.jelly](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/lib/form/hetero-list.jelly) is configured with both `oneEach=true` and `honorOrder=true`. This this might not be only publishers [who use this](https://github.com/search?p=1&q=org%3Ajenkinsci+honorOrder+oneEach&ref=searchresults&type=Code&utf8=%E2%9C%93), I suggest to trigger this feature based on those classes instead.

:+1:  on using descriptorId fingerprint.